docker/misk-web/misk-web

```diff
@@ -64,15 +62,28 @@ hashDir() {
 }
 
 hashDiff() {
+    oldHash=$(echo "$1" | sed 's/[[:space:]]//g')
+    newHash=$(echo "$2" | sed 's/[[:space:]]//g')
+    if [ "$oldHash" = "$newHash" ]; then
+        echo "${oldHash} ${newHash} false"
+    else
+        echo "${oldHash} ${newHash} true"
+    fi
+}
+
+hashDiffFiles() {
+    oldHashFile="$1"
+    newHashFile="$2"
+    [ -f "$oldHashFile" ] || echo "true"
```

Comment:
What are these two lines doing? Early exit with true if either doesn't exist?

Reply:
Correct. If the oldHash file `$(pwd)/.hash` doesn't exist, then the comparison of the directory contents can't be checked at all so automatically falsify the existing cache.

Reply:
Seems like this may echo "hash1 hash2 true/false" or just "true" depending on whether or not both files exist. Is that fine?